sklearn/compose/tests/test_column_transformer.py

```diff
@@ -551,12 +552,15 @@ def test_column_transformer_remainder_numpy(key):
     assert_array_equal(ct.fit(X_array).transform(X_array), X_res_both)
 
 
-@pytest.mark.parametrize("key", [[0], slice(0, 1), np.array([True, False]),
-                                 ['first'], slice(None, 'first'),
-                                 slice('first', 'first')])
+@pytest.mark.parametrize(
+    "key", [[0], slice(0, 1), np.array([True, False]), ['first'], 'pd-index',
```

Comment:
Such param lists I find are easier to read (and certainly to diff) if each value is on its own line. Is pd-index the only difference? Can you explain how this is equivalent to the reported issue?

Reply:
No, the `np.array(['first']), np.array(['first'], dtype=object)` is also new (in addition to `pd-index`).

The selection of the columns worked for those column-specifications, but it was the determination of the remaining columns that failed for those. That is what this test is about, and thus adding those cases to test that the remaining is working.

Reply:
I think parameters can be marked with skip-if tags, but you'd still need to not construction, so fair enough.

Reply:
I see.